Objects/listobject.c

```diff
@@ -180,6 +180,22 @@ PyList_New(Py_ssize_t size)
     return (PyObject *) op;
 }
 
+static inline PyObject *
+list_new(Py_ssize_t size) {
+    PyListObject *op = (PyListObject *) PyList_New(0);
+    if (size == 0 || op == NULL) {
+        return (PyObject *) op;
+    }
+    op->ob_item = (PyObject **) PyMem_Malloc(size * sizeof(PyObject *));
```

Comment:
You should check if size*sizeof() overflow or not.

You might allocate ob_item before creating op, since I dislike calling Py_DECREF() on a just created object (is it fully initialized or not?).

Reply:
Especially, in this case deallocating the list will decref uninitialize list items...

Reply:
> You might allocate ob_item before creating op, since I dislike calling Py_DECREF() on a just created object (is it fully initialized or not?).

Could you explain the difference?

Reply:
> Especially, in this case deallocating the list will decref uninitialized list items

No, it won't. Either the `PyMem_Malloc()` fails, then `ob_item` is NULL and the list size is 0, which is consistent and can be cleaned up safely. Or, if `PyMem_Malloc()` succeeds, then `ob_item` points to the item array, which will be filled up right afterwards.

I think this code is ok as it is.